generator/templates/server/parameter.gotmpl

```diff
@@ -120,11 +120,17 @@ type {{ pascalize .Name }}Params struct {
 func ({{ .ReceiverName }} *{{ pascalize .Name }}Params) BindRequest(r *http.Request, route *middleware.MatchedRoute) error {
   var res []error
   {{ if .HasQueryParams }}qs := httpkit.Values(r.URL.Query())
-  {{ else if .HasFormParams }}if err := r.ParseMultipartForm(32 << 20); err != nil {
-		return err
-	}
-	fds := httpkit.Values(r.Form)
-  {{ end }}
+  {{ else if .HasFormParams }}if r.Header.Get("Content-Type") == "multipart/form-data" {
```

Comment:
should this be a call to a helper function in httpkit that parses the header into a media type format?
I think at the very least it wants to be a prefix check.